core/reactor_common.c

```diff
@@ -358,10 +362,18 @@ static token_freed _lf_free_token(lf_token_t* token) {
     token_freed result = NOT_FREED;
     if (token->value != NULL) {
         // Count frees to issue a warning if this is never freed.
-        // Do not free the value field if it is garbage collected and token's ok_to_free field is not "token_and_value".
+        // Do not free the value field if it is garbage collected and token's
+        // ok_to_free field is not "token_and_value".
         _lf_count_payload_allocations--;
-        if (OK_TO_FREE != token_only && token->ok_to_free == token_and_value) {
-            LF_PRINT_DEBUG("_lf_free_token: Freeing allocated memory for payload (token value): %p", token->value);
+        // Free the value field (the payload).
+        // First check whether the value field is garbage collected (e.g. in the
+        // Python target), in which case the payload should not be freed.
+        if (OK_TO_FREE != token_only
+                && token->value != NULL
+                && token->ok_to_free != token_only
+            ) {
```

Comment:
Is there a difference between `token->ok_to_free == token_and_value` and `!= token_only`?